### PR TITLE
Task5 View articles by Id

### PR DESCRIPTION
**Pull request by stevenwu99**
I have completed Task 5 for viewing the article by Id.
Please check.
Thank you

### CODE REVIEW

File: src/components/ShowArticlesById.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    const [article,setArticle] = useState([]);
+    const [isLoading,setIsLoading] = useState(true)
+    const [errMsg,setErrMsg] = useState("")
+   
+    let  searchUrl = "";
+
+     if (articleId !== "") {
+        const articleIdNum = Number(articleId)
+        searchUrl = `/articles/${articleIdNum}`;
```
Comment (popatre) [inline src/components/ShowArticlesById.jsx:18]: You can use useParams to get the url

File: src/components/ShowArticlesById.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        console.log('Search URL-->',searchUrl)
+    } 
+
+     useEffect (() =>{
+       if (searchUrl !== "") {
+           getData(searchUrl)
```
Comment (popatre) [inline src/components/ShowArticlesById.jsx:23]: Not sure what getData is naming wise

**Review by popatre — CHANGES_REQUESTED** [review]
- Excellent see some loading feedback to let us know something is happening

Quite a few changes needed please Steven,
- [x] I should be able to navigate to the article via the url e.g. article/1 and get the correct data. At the moment you have to type in the id. A user of the website wouldnt really know what id of the article they want is. Ideally, we should be able to click the list of articles and let it navigate us to the correct article id
- [ ] Please render the relevant article from the api - we should have a page that has some more details about the article, like its body.
- [x] Need to use the article id to fetch the data when someone navigates to the page
- [x] Might be nicer to have the url as /articles/:article_id - you can then use useParams to get the article_id and do your api request. I should be able to share the page with someone, but at the moment it isnt there when i refresh.

File: src/components/ShowArticlesById.jsx (hunk 2 above)
Comment (stevenwu99) [reply src/components/ShowArticlesById.jsx:23]: This function is a reuse function. I want all the GET methods will use getData to get the data from API.

File: src/components/ShowArticlesById.jsx (hunk 1 above)
Comment (stevenwu99) [reply src/components/ShowArticlesById.jsx:18]: Done

**Review by popatre — CHANGES_REQUESTED** [review]
Looks much better Steven! Well done. 

One small thing to add please

- [ ] The article data from the api should have a key of `body` - which has some more information on the article. Could we display that body information as well as the other infor you've currently got?

**Pull request merged**